plugins/inventory/terraform_state.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+DOCUMENTATION = r"""
+name: terraform_state
+author:
+  - Aubin Bikouo (@abikouo)
+short_description: Builds an inventory from resources created by cloud providers.
+description:
+  - This plugin works with an existing state file to create an inventory from resources created by cloud providers.
+  - The plugin accepts a Terraform backend config to an existing state file or a path to an existing state file.
+  - To read the state file command ``Terraform show`` is used.
+  - Does not support caching.
+extends_documentation_fragment:
+  - constructed
+version_added: 2.1.0
+options:
+  plugin:
+    description:
+      - The name of the Inventory Plugin.
+      - This should always be C(cloud.terraform.terraform_state).
+    required: true
+    type: str
+    choices: [ cloud.terraform.terraform_state ]
+  backend_config:
+    description:
+      - A Terraform backend configuration to an existing state file.
+    type: str
+  search_child_modules:
+    description:
+      - Whether to include resources from Terraform child modules.
+    type: bool
+    default: false
+  binary_path:
+    description:
+      - The path of a terraform binary to use.
+    type: path
+  hostnames:
+    description:
+      - A list in order of precedence for hostname variables.
+      - The elements of the list can be a dict with the keys mentioned below or a string.
+      - Can be one of the options specified in U(https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#argument-reference).
+      - If value provided does not exist in the above options, it will be used as a literal string.
+      - To use tags as hostnames use the syntax tag:Name=Value to use the hostname Name_Value, or tag:Name to use the value of the Name tag.
+      - If not provided the final hostname will be C(terraform resource type) + C(_) + C(terraform resource name)
+    type: list
+    elements: raw
+    default: []
+    suboptions:
+      name:
+        description:
+          - Name of the host.
+        type: str
+        required: True
+      prefix:
+        description:
+          - Prefix to prepend to I(name). Same options as I(name).
+          - If I(prefix) is specified, final hostname will be I(prefix) +  I(separator) + I(name).
+        type: str
+        default: ''
+        required: False
+      separator:
+        description:
+          - Value to separate I(prefix) and I(name) when I(prefix) is specified.
+        type: str
+        default: '_'
+        required: False
+"""
+
+EXAMPLES = r"""
+# Example using constructed features to set ansible_host
+plugin: cloud.terraform.terraform_state
+backend_config: |
+  backend "http" {
+      address = "https://localhost:8043/api/v2/state/3/"
+      skip_cert_verification = true
+      username = "ansible"
+      password = "test123!"
+  }
+compose:
```

Comment:
If we are showing an example with `compose` should we also show an example with `keyed_groups`?

Reply:
done